.circleci/config.yml

```diff
@@ -39,8 +39,6 @@ jobs:
     working_directory: *workspace_root
     steps:
       - checkout
-      - tools/install-golangci-lint:
-          version: 1.33.2
```

Comment:
Do we still want to control the golangci-lint version? I'm wondering if it's valuable to keep the version in sync with the one used in plugins. https://github.com/mattermost/circleci-orbs/blob/ea9f34540218ce3a62b101eab2c4fffc331be11b/plugin-ci/orb.yml#L51-L53

Reply:
3/5 it's, in general, helpful to use the same `golangci-lint` version across multiple projects as there are fewer "surprises" when CI runs. A dev can only install one version locally, and that should ideally match one used in CI. For (most) plugins, the centralised CI config already synchronises the version used.

On the other hand, it's cumbersome to update the version for all other projects at the same time. Hence, I suggested we use this opportunity to sync the version with the one used in the plugin repos.

Reply:
@hanzei Thanks! 

Ok, I figured how to get the up to date `golangci-lint` -- it's included in the `cimg/go` container already, so no need for the `gotest/tools`. Probably the `plugin-ci` orb can be simplified (we don't need to install `golangci-lint` there either)

Reply:
I'll merge now so I can start with the production branch merge.